lib/vtls/openssl.c

```diff
@@ -2715,10 +2721,51 @@ static CURLcode ossl_connect_step1(struct connectdata *conn, int sockindex)
   }
 #endif
 
+
+#if defined(USE_WIN32_CRYPTO) && \
+        !defined(LIBCURL_WINDOWS_OPENSSL_NEVER_USE_WINDOWS_CA_STORE)
+  /* read https://stackoverflow.com/questions/9507184/ and
+  https://github.com/d3x0r/SACK/blob/master/src/netlib/ssl_layer.c#L1037 */
+  if(ssl_cafile && (strcmp(ssl_cafile, CURL_WINDOWS_CA_STORE) == 0) &&
+      (!ssl_capath)) {
+
+    X509_STORE* store = SSL_CTX_get_cert_store(BACKEND->ctx);
+    PCCERT_CONTEXT pContext = NULL;
+
+    HCERTSTORE hStore = CertOpenSystemStoreA((HCRYPTPROV_LEGACY)NULL,
+        "ROOT");
+
+    if(hStore) {
+      while((pContext = CertEnumCertificatesInStore(hStore, pContext))
```

Comment:
do any attributes of the certificate have to be checked like notbefore notafter etc or does openssl do this